### DISCUSSION TITLE
Is there a limit to how large history.db can be?

### DISCUSSION THREAD

**Discussion by vF360** [FreeTubeApp/FreeTube#5682] [Q&A] [answered]
I'm thinking about turning on history as youtubers seem to keep changing the title & thumbnail of their videos, making it hard for me to figure out if I've seen certain videos before or not. However, I don't want to run into an issue in like 5-10 years where I need to start over from scratch because history.db is too large and needs to be reset, which is why I am hesistant to commit to it. 

FreeTube could solve this problem by splitting history.db up into multiple files once they cross a certain size limit instead of cramming everything into 1 place, if it even is an issue and not just me being paranoid.

**Comment by absidue** [comment]
We've never tested it but the limit is less likely to be a problem with the size of the file and more with the quantity of videos and splitting the files won't help with that. FreeTube just like most apps will experience slow downs when it has to deal with gigantic amounts of data. However I suspect that it won't be a problem for a long time.

Long story short: Don't worry about it and if it does become a problem in the future you can delete old history entries to reduce the quantity again.

**Reply by vF360** [reply]
> We've never tested it but the limit is less likely to be a problem with the size of the file and more with the quantity of videos and splitting the files won't help with that. FreeTube just like most apps will experience slow downs when it has to deal with gigantic amounts of data. However I suspect that it won't be a problem for a long time.
> 
> Long story short: Don't worry about it and if it does become a problem in the future you can delete old history entries to reduce the quantity again.

Deleting old history entries defeats the entire purpose as the whole point of logging the videos I've seen is so that I don't re-watch the same stuff. There is no way that the file will be larger than a few gigabites after a lifetime of usage, so I don't really see how 'slow downs' can occur, especially if it a local file.

**Reply by absidue** [reply, marked as answer]
It's not about the file size but the number of videos and yes large quantities of data will most certainly lead to slow downs in FreeTube. There is literally an open issue about large playlists freezing the app for multiple seconds and historical ones about large watch histories causing performance problems on the subscriptions page.
